quickwit-search/src/collector.rs

```diff
@@ -64,7 +68,12 @@ impl SortingFieldComputer {
                     SortOrder::Asc => u64::MAX - field_val,
                 }
             }
-            SortingFieldComputer::SortByDocId => 0u64,
+            SortingFieldComputer::DocId => 0u64,
+            SortingFieldComputer::Score { order } => match order {
+                // We lose some precision in order to properly cast the f32 to u64
+                SortOrder::Desc => score.mul(1000.0) as u64,
```

Comment:
That works, but just for fun:
https://lemire.me/blog/2020/12/14/converting-floating-point-numbers-to-integers-while-preserving-order/